frontend/ui/widget/htmlboxwidget.lua

```diff
@@ -103,6 +109,12 @@ function HtmlBoxWidget:_render()
     local dc = DrawContext.new()
     self.bb = page:draw_new(dc, self.dimen.w, self.dimen.h, 0, 0)
     page:close()
+
+    if self.highlight_rects then
+        for _, rect in ipairs(self.highlight_rects) do
+            self.bb:invertRect(rect.x, rect.y, rect.w, rect.h)
+        end
```

Comment:
I did not notice any horizontal overlaps, but there is vertical overlap that adds a white pixel line between the lines.

I won't use koptinterface:getTextFromBoxes because it doesn't work correctly for right to left text. I will modify the original HtmlBoxWidget:getSelectedText that I removed, to get the rectangles the same way as in TextBoxWidget. The way that works ensures that there are no horizontal overlaps.

As for the vertical ones: what do you think, should the highlighting use the same grey rectangle instead?

Reply:
> but there is vertical overlap that adds a white pixel line between the lines.

That may depends on the CSS/styling of the HTML content ? Our default line-height or the dict publisher/lua func tweaks.

> As for the vertical ones: what do you think, should the highlighting use the same grey rectangle instead?

No real thought :) I haven't yet seen how it looks. Not fond of reverse video, but it may be easier (to invert and restore) than our grey/dim/lighten stuff.
Also, dunno how reverse/invert would do on HTML content with foreground and background colors. I have a dict + luafunc that displays stuff like this:
![image](https://github.com/user-attachments/assets/fddb25e0-9e0b-40d0-a5f6-9ce3d9089a03)